DataConnectors/AtlassianConfluenceAudit/ConfluenceAuditAPISentinelConnector/__init__.py

```diff
@@ -0,0 +1,126 @@
+import requests
+import json
+import datetime
+from requests.auth import HTTPBasicAuth
+import azure.functions as func
+import base64
+import hmac
+import hashlib
+import os
+import logging
+import re
+from .state_manager import StateManager
+
+customer_id = os.environ['WorkspaceID']
+shared_key = os.environ['WorkspaceKey']
+confluence_token = os.environ['ConfluenceAccessToken']
+confluence_username = os.environ['ConfluenceUsername']
+confluence_homesite_name = os.environ['ConfluenceHomeSiteName']
+connection_string = os.environ['AzureWebJobsStorage']
+log_type = 'Confluence_Audit'
+confluence_uri_audit = "https://" + confluence_homesite_name + ".atlassian.net/wiki/rest/api/audit"
+
+if 'logAnalyticsUri' in os.environ:

```

Comment:
@averbn ,if log logAnalyticsUri values is null, in that case it will fail to build default logAnalyticsUri. Can you please update following code to handle this scenario. Thanks!

logAnalyticsUri = os.environ.get('logAnalyticsUri')
if ((logAnalyticsUri in (None, '') or str(logAnalyticsUri).isspace())):    
    logAnalyticsUri = 'https://' + customerId + '.ods.opinsights.azure.com'

pattern = r"https:\/\/([\w\-]+)\.ods\.opinsights\.azure.([a-zA-Z\.]+)$"
match = re.match(pattern,str(logAnalyticsUri))
if(not match):
    raise Exception("Invalid Log Analytics Uri.")

Please refer the sample template : https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/Templates/Connector_REST_API_AzureFunctionApp_template/Template_REST_API_AzureFunction_App_Code/Template_REST_API_Function_App_Python/Template_REST_API_Function_App_Python.py